File: backend/models/request.py
from . import db
from datetime import datetime

class Request(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    type = db.Column(db.Enum('rent', 'buy', 'trade-in'), nullable=False)
    status = db.Column(db.Enum('new', 'pending', 'on-processing','done'), nullable=False)
    user_id = db.Column(db.Integer, db.ForeignKey('user.id'), nullable=False)
    car_id = db.Column(db.Integer, db.ForeignKey('car.id'), nullable=False)
    start_date = db.Column(db.Date, nullable=False)
    end_date = db.Column(db.Date)
    created_at = db.Column(db.DateTime, default=datetime.utcnow)
    price = db.Column(db.Numeric(10, 2))
    description = db.Column(db.Text)

File: frontend/src/Components/types/request.ts
export interface Request {
    id: number;
    type: string;
    status: string;
    user_id: number;
    car_id: number;
    start_date: string;
    end_date?: string;
    created_at: string;
    price?: number;
    description?: string;
}

export interface RequestResponse {
    success: boolean;
    data?: Request | Request[];
    message?: string;
} 